### PR TITLE
Closes #106 | Create dataset loader for NewsPH

### LINKED ISSUE #106: Create dataset loader for NewsPH
Dataloader name: `newsph/newsph.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?newsph

|     Dataset| newsph |
|-------------|---|
| Description | Raw collection of news articles in Filipino which can be used for language modelling. |
| Subsets | - |
| Languages | tgl, fil |
| Tasks | Language Modeling |
| License | GNU General Public License v3.0 (gpl-3.0) |
| Homepage | https://huggingface.co/datasets/newsph |
| HF URL | https://huggingface.co/datasets/newsph |
| Paper URL | https://arxiv.org/abs/2010.11574 |

### PR DESCRIPTION
**Pull request by IvanHalimP**
Closes #106 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/newsph/newsph.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """\
+@article{cruz2020investigating,
+  title={Investigating the True Performance of Transformers in Low-Resource Languages: A Case Study in Automatic Corpus Creation},
+  author={Jan Christian Blaise Cruz and Jose Kristian Resabal and James Lin and Dan John Velasco and Charibeth Cheng},
+  journal={arXiv preprint arXiv:2010.11574},
+  year={2020}
+}
+"""
```
Comment (jamesjaya) [inline seacrowd/sea_datasets/newsph/newsph.py:19]: `_CITATION` here uses the old version of the paper described by the author here https://github.com/jcblaisecruz02/Filipino-Text-Benchmarks#citations which was last updated 3 years ago, on the other hand the bibtex export from the provided arxiv URL on the issue ticket uses the new title from the 2021 version. I don't know which one we need to use here. Maybe it would be great if you could check with Blaise Cruz if needed.

```
@misc{cruz2021exploiting,
      title={Exploiting News Article Structure for Automatic Corpus Generation of Entailment Datasets}, 
      author={Jan Christian Blaise Cruz and Jose Kristian Resabal and James Lin and Dan John Velasco and Charibeth Cheng},
      year={2021},
      eprint={2010.11574},
      archivePrefix={arXiv},
      primaryClass={cs.CL}
}
```

File: seacrowd/sea_datasets/newsph/newsph.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        SEACrowdConfig(
+            name="newsph_seacrowd_pairs",
+            version=SEACROWD_VERSION,
+            description="newsph SEACrowd schema",
+            schema="seacrowd_pairs",
+            subset_id="newsph",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "newsph_source"
```
Comment (jamesjaya) [inline seacrowd/sea_datasets/newsph/newsph.py:60]: I don't know if we should enforce this but other reviewers seem to favor the usage of _DATASETNAME constant in configs even though it is not explicitly prescribed in the guidelines. Personally I'm okay with this and I think it's cleaner this way.

**Review by jamesjaya — APPROVED** [review]
tested, looks good to me, but I have some remarks which I can't decide on

File: seacrowd/sea_datasets/newsph/newsph.py (hunk 1 above)
Comment (holylovenia) [reply seacrowd/sea_datasets/newsph/newsph.py:19]: Hi @jcblaisecruz02, could you please confirm this as the original author? @IvanHalimP can update his dataloader and I can update the datasheet accordingly.

Thanks for raising this question, @jamesjaya.

File: seacrowd/sea_datasets/newsph/newsph.py (hunk 2 above)
Comment (holylovenia) [reply seacrowd/sea_datasets/newsph/newsph.py:60]: I'm okay with this as well as long as the output is correct, so no problem.

File: seacrowd/sea_datasets/newsph/newsph.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+"""
+_DATASETNAME = "newsph"
+_LANGS = ["fil", "tgl"]
+_DESCRIPTION = """\
+Raw collection of news articles in Filipino which can be used for language modelling.
+"""
+_HOMEPAGE = "https://huggingface.co/datasets/newsph"
+_LICENSE = Licenses.GPL_3_0.value
+_LOCAL = False
+_URLS = "https://s3.us-east-2.amazonaws.com/blaisecruz.com/datasets/newsph/newsph-nli.zip"
```
Comment (holylovenia) [inline seacrowd/sea_datasets/newsph/newsph.py:28]: @IvanHalimP I think you used the wrong data URL. This dataloader is supposedly for [NewsPH](https://huggingface.co/datasets/newsph), which is a language modeling task, as provided in the #106. The URL that you used is supposedly for NewsPH-NLI, a derivative of NewsPH.

May I suggest you to:
1. Use the `LANGUAGE_MODELING` task and [this data URL](https://huggingface.co/datasets/newsph) for NewsPH dataloader.
2. Submit the NewsPH-NLI datasheet via [our form](https://form.jotform.com/team/232952680898069/seacrowd-sea-datasets).
3. Make a pull request using the current dataloader for NewsPH-NLI after the issue comes up (+ modify some of the constants, e.g., `_DESCRIPTION`, `_SUPPORTED_TASKS`, etc., accordingly)?

Please let me know what you think and update me on your next course of action so I can help.

**Review by holylovenia — CHANGES_REQUESTED** [review]

File: seacrowd/sea_datasets/newsph/newsph.py (hunk 1 above)
Comment (jcblaisecruz02) [reply seacrowd/sea_datasets/newsph/newsph.py:19]: Hi! @holylovenia @jamesjaya 

The 2021 (published) version uses the title “**Exploiting News Article Structure for Automatic Corpus Generation of Entailment Datasets**” and should be the correct one moving forward

The correct bibtex entry is thus:
```
@inproceedings{cruz2021exploiting,
  title={Exploiting news article structure for automatic corpus generation of entailment datasets},
  author={Cruz, Jan Christian Blaise and Resabal, Jose Kristian and Lin, James and Velasco, Dan John and Cheng, Charibeth},
  booktitle={PRICAI 2021: Trends in Artificial Intelligence: 18th Pacific Rim International Conference on Artificial Intelligence, PRICAI 2021, Hanoi, Vietnam, November 8--12, 2021, Proceedings, Part II 18},
  pages={86--99},
  year={2021},
  organization={Springer}
}
```
which is also what you get when searching the citation via Google Scholar.

Sorry for the confusion! We’ve been meaning to update that github repo for some time but it’s been buried under other to-dos.

File: seacrowd/sea_datasets/newsph/newsph.py (hunk 3 above)
Comment (IvanHalimP) [reply seacrowd/sea_datasets/newsph/newsph.py:28]: Hi all, thank you for the comments. 
Please kindly wait for the time being as I need to take care of something else.

Comment (holylovenia) [reply seacrowd/sea_datasets/newsph/newsph.py:28]: @IvanHalimP Sure sure, just let us know if we can help with anything. Enjoy your holiday~

File: seacrowd/sea_datasets/newsph/newsph.py (hunk 1 above)
Comment (IvanHalimP) [reply seacrowd/sea_datasets/newsph/newsph.py:19]: ok

File: seacrowd/sea_datasets/newsph/newsph.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+_CITATION = """\
+@article{cruz2020investigating,
+  title={Investigating the True Performance of Transformers in Low-Resource Languages: A Case Study in Automatic Corpus Creation},
+  author={Jan Christian Blaise Cruz and Jose Kristian Resabal and James Lin and Dan John Velasco and Charibeth Cheng},
+  journal={arXiv preprint arXiv:2010.11574},
+  year={2020}
+}
+"""
+_DATASETNAME = "newsph"
+_LANGS = ["fil", "tgl"]
```
[the remark this answers is not in the document]
Comment (IvanHalimP) [reply seacrowd/sea_datasets/newsph/newsph.py:21]: ok

File: seacrowd/sea_datasets/newsph/newsph.py (hunk 3 above)
Comment (IvanHalimP) [reply seacrowd/sea_datasets/newsph/newsph.py:28]: Regarding the related task for this newsph dataset, there is no labels and no pairs. AFAIK, it falls into self-supervised pretraining schema then?

Comment (holylovenia) [reply seacrowd/sea_datasets/newsph/newsph.py]: @IvanHalimP Yes yes. I saw your changes and tested them. The dataloader works well on my end. Thanks for the modifications. I'm merging it now.
